components/org.wso2.carbon.identity.oauth.dcr/src/test/java/org/wso2/carbon/identity/oauth/dcr/factory/RegistrationRequestFactoryTest.java

```diff
@@ -0,0 +1,257 @@
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+import org.wso2.carbon.base.CarbonBaseConstants;
+import org.wso2.carbon.context.PrivilegedCarbonContext;
+import org.wso2.carbon.identity.application.authentication.framework.inbound.FrameworkClientException;
+import org.wso2.carbon.identity.application.authentication.framework.inbound.HttpIdentityRequestFactory;
+import org.wso2.carbon.identity.application.authentication.framework.inbound.HttpIdentityResponse;
+import org.wso2.carbon.identity.oauth.dcr.model.RegistrationRequest;
+
+import java.io.BufferedReader;
+import java.nio.file.Paths;
+
+import org.wso2.carbon.identity.oauth.dcr.model.RegistrationRequestProfile;
+import org.wso2.carbon.identity.testutil.powermock.PowerMockIdentityBaseTest;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.ws.rs.core.MediaType;
+
+import static org.mockito.Matchers.anyInt;
+import static org.mockito.Matchers.anyString;
+import static org.powermock.api.mockito.PowerMockito.when;
+import static org.powermock.api.mockito.PowerMockito.mock;
+import static org.powermock.api.mockito.PowerMockito.whenNew;
+import static org.powermock.api.mockito.PowerMockito.doAnswer;
+
+import static org.powermock.api.support.membermodification.MemberMatcher.methodsDeclaredIn;
+import static org.powermock.api.support.membermodification.MemberModifier.suppress;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.fail;
+
+@PrepareForTest(RegistrationRequestFactory.class)
+public class RegistrationRequestFactoryTest extends PowerMockIdentityBaseTest {
+
+    private RegistrationRequestFactory registrationRequestFactory;
+    private String dummyDescription = "dummyDescription";
+    private String ownerName = "dummyOwnerName";
+
+    @Mock
+    HttpServletRequest mockHttpRequest;
+
+    @Mock
+    HttpServletResponse mockHttpResponse;
+
+    @Mock
+    RegistrationRequest.RegistrationRequestBuilder mockRegistrationRequestBuilder;
+
+    @Mock
+    HttpIdentityResponse.HttpIdentityResponseBuilder mockHttpIdentityResponseBuilder;
+
+    @Mock
+    BufferedReader mockReader;
+
+    @Mock
+    JSONParser jsonParser;
+
+    @BeforeMethod
+    private void setUp() {
+
+        registrationRequestFactory = new RegistrationRequestFactory();
+    }
+
+    @DataProvider(name = "JSONOBJECTDataProvider")
+    public Object[][] getData() {
+
+        JSONArray grantTypes = new JSONArray();
+        JSONArray redirectUrls = new JSONArray();
+        JSONArray responseTypes = new JSONArray();
+        JSONArray scopes = new JSONArray();
+        JSONArray contacts = new JSONArray();
+        grantTypes.add("dummyGrantType");
+        redirectUrls.add("dummyRedirectUrl");
+        responseTypes.add("dummyResponseType");
+        contacts.add("dummyContact");
+        scopes.add("dummyScope");
+
+        String grantType = "dummyGrantType";
```

Comment:
Why you define constant here and use plain string in line [81](https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/574/files#diff-ea882f3939d7b33008446ae944e7fa5fR81) ?